Zotero/Scenes/Detail/SingleCitation/Views/SingleCitationViewController.swift

```diff
@@ -131,8 +138,9 @@ final class SingleCitationViewController: UIViewController {
         updatePreferredContentSize()
     }
 
-    deinit {
-        viewModel.process(action: .cleanup)
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        previewWebView.configuration.userContentController.removeAllScriptMessageHandlers()
```

Comment:
Would it be ok to do this in `deinit` too? These handlers are setup in `viewDidLoad` and removed here in `viewWillDisappear`. So if this view disappears and re-appears again they won't work. I know that UI probably doesn't allow that now, but if we add something later it might cause an issue.

Reply:
This needs to be called before, so that the reference is removed, in order to trigger deinitialization. We can move the handler addition in `viewWillAppear`, so that they are always balanced. I tested it, and it works as expected. 
Check https://github.com/zotero/zotero-ios/pull/831/commits/6e44151ba8a5b15455e029416b3a08976d9aad23